store/tikv/tikvrpc/tikvrpc.go

```diff
@@ -354,3 +372,138 @@ func (resp *Response) GetRegionError() (*errorpb.Error, error) {
+	resp := &Response{}
+	resp.Type = req.Type
+	var err error
+	switch req.Type {
+	case CmdGet:
+		resp.Get, err = client.KvGet(ctx, req.Get)
+	case CmdScan:
+		resp.Scan, err = client.KvScan(ctx, req.Scan)
+	case CmdPrewrite:
+		resp.Prewrite, err = client.KvPrewrite(ctx, req.Prewrite)
+	case CmdCommit:
+		resp.Commit, err = client.KvCommit(ctx, req.Commit)
+	case CmdCleanup:
+		resp.Cleanup, err = client.KvCleanup(ctx, req.Cleanup)
+	case CmdBatchGet:
+		resp.BatchGet, err = client.KvBatchGet(ctx, req.BatchGet)
+	case CmdBatchRollback:
+		resp.BatchRollback, err = client.KvBatchRollback(ctx, req.BatchRollback)
+	case CmdScanLock:
+		resp.ScanLock, err = client.KvScanLock(ctx, req.ScanLock)
+	case CmdResolveLock:
+		resp.ResolveLock, err = client.KvResolveLock(ctx, req.ResolveLock)
+	case CmdGC:
+		resp.GC, err = client.KvGC(ctx, req.GC)
+	case CmdDeleteRange:
+		resp.DeleteRange, err = client.KvDeleteRange(ctx, req.DeleteRange)
+	case CmdRawGet:
+		resp.RawGet, err = client.RawGet(ctx, req.RawGet)
+	case CmdRawPut:
+		resp.RawPut, err = client.RawPut(ctx, req.RawPut)
+	case CmdRawDelete:
+		resp.RawDelete, err = client.RawDelete(ctx, req.RawDelete)
+	case CmdRawScan:
+		resp.RawScan, err = client.RawScan(ctx, req.RawScan)
+	case CmdCop:
+		resp.Cop, err = client.Coprocessor(ctx, req.Cop)
+	case CmdCopStream:
+		// Use context to support timeout for grpc streaming client.
+		ctx1, cancel := context.WithCancel(ctx)
+		var stream tikvpb.Tikv_CoprocessorStreamClient
+		stream, err = client.CoprocessorStream(ctx1, req.Cop)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		wrapStream := NewCopStreamResponse(stream, nil, cancel, ch)
+		// Read the first streaming response to get CopStreamResponse.
+		// This can make error handling much easier, because SendReq() retry on
+		// region error automatically.
+		var first *coprocessor.Response
+		first, err = wrapStream.Recv()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		wrapStream.Response = first
+		resp.CopStream = wrapStream
+	case CmdMvccGetByKey:
+		resp.MvccGetByKey, err = client.MvccGetByKey(ctx, req.MvccGetByKey)
+	case CmdMvccGetByStartTs:
+		resp.MvccGetByStartTS, err = client.MvccGetByStartTs(ctx, req.MvccGetByStartTs)
+	case CmdSplitRegion:
+		resp.SplitRegion, err = client.SplitRegion(ctx, req.SplitRegion)
+	default:
+		return nil, errors.Errorf("invalid request type: %v", req.Type)
+	}
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return resp, nil
+}
+
+// TimeoutItem is used to implement grpc stream timeout.
+type TimeoutItem struct {
+	cancel   context.CancelFunc
+	timeout  time.Time
+	complete int32 // atomic variable, 0 means false
+}
+
+// Recv overrides the stream client Recv() function.
+func (resp *CopStreamResponse) Recv() (*coprocessor.Response, error) {
+	item := &TimeoutItem{
+		cancel: resp.cancel,
+		// TODO: Make timeout a field in CopStreamResponse.
+		timeout: time.Now().Add(20 * time.Second),
```

Comment:
Use a constant instead of hard code?

Reply:
Just keep it here, I'd like to fix it in another PR, tiny refactor the code related to set timeout.